elasticsearch/src/main/java/org/apache/calcite/adapter/elasticsearch/ElasticsearchSchemaFactory.java

```diff
@@ -56,14 +59,23 @@ public ElasticsearchSchemaFactory() {
 
     try {
       final String coordinatesString = (String) map.get("coordinates");
-      Preconditions.checkState(coordinatesString != null,
-          "'coordinates' is missing in configuration");
 
-      final Map<String, Integer> coordinates = mapper.readValue(coordinatesString,
-          new TypeReference<Map<String, Integer>>() { });
+      final String hostsString = (String) map.get("hosts");
+      Preconditions.checkState(coordinatesString != null || hostsString != null,
+          "Both 'coordinates' and 'hosts' is missing in configuration. Provide one of them.");
 
       // create client
-      final RestClient client = connect(coordinates);
+      final RestClient client;
+
+      if (coordinatesString != null) {
+        final Map<String, Integer> coordinates = mapper.readValue(coordinatesString,
```

Comment:
Are there 2 versions of `connect` function  ? 
I would rather create `List<String>` (or `List<HttpHost>`) here and pass it to unique `connect` method.
Don't forget to use [HttpHost.create()](https://hc.apache.org/httpcomponents-core-ga/httpcore/apidocs/org/apache/http/HttpHost.html#create(java.lang.String)) instead of URL

Reply:
Done